spring-integration-test/src/main/java/org/springframework/integration/test/context/SpringIntegrationTestExecutionListener.java

```diff
@@ -66,4 +75,15 @@ private boolean match(String name, String[] patterns) {
 						.anyMatch(pattern -> PatternMatchUtils.simpleMatch(pattern, name));
 	}
 
+	/**
+	 * to allow nested class execution
+	 */
+	private MockIntegrationContext handledGetMockIntegrationContext(ApplicationContext applicationContext) {
```

Comment:
See the `MockIntegrationContextCustomizerFactory` for similar `TestContextAnnotationUtils` fix instead.

Reply:
Got it [here](https://github.com/spring-projects/spring-integration/pull/9506/files#diff-7c2fcf66d0bd0e26c5966321b6dd529e60067a1d1c36200d3ca82fdabbeb6e47R42)